weblate/api/views.py

```diff
@@ -726,8 +731,53 @@ def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
 
 
+class CreditsMixin:
+    @action(detail=True, methods=["get"])
+    def credits(self, request, **kwargs):
+        if request.user.is_anonymous:
+            self.permission_denied(request, "Must be authenticated to get credits")
+
+        obj = self.get_object()
+
+        try:
+            start_date = from_current_timezone(
+                datetime.fromisoformat(request.query_params["start"])
+            )
+        except (ValueError, MultiValueDictKeyError) as err:
+            raise BadRequest("Invalid format for `start`") from err
+
+        try:
+            end_date = from_current_timezone(
+                datetime.fromisoformat(request.query_params["end"])
+            )
+        except (ValueError, MultiValueDictKeyError) as err:
+            raise BadRequest("Invalid format for `end`") from err
+
+        language = None
+
+        if "lang" in request.query_params:
+            language = request.query_params["lang"]
+
+        kwargs: dict[str, Any]
+        if isinstance(obj, Project):
+            kwargs = {"translation__component__project": obj}
+        elif isinstance(obj, Component):
+            kwargs = {"translation__component": obj}
+        else:
+            raise TypeError("Expected project or component")
```

Comment:
Maybe we can push this logic to `generate_credits` so that it doesn't have to be duplicated?